exonum-java-binding/core/src/test/java/com/exonum/binding/core/runtime/TransactionInvokerTest.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2019 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.core.runtime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.core.service.Node;
+import com.exonum.binding.core.service.Service;
+import com.exonum.binding.core.storage.database.Snapshot;
+import com.exonum.binding.core.transaction.TransactionContext;
+import com.exonum.binding.core.transaction.TransactionExecutionException;
+import com.exonum.binding.core.transaction.TransactionMethod;
+import io.vertx.ext.web.Router;
+import java.util.Collections;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+
+class TransactionInvokerTest {
+
+  private static final byte[] ARGUMENTS = new byte[0];
+  @Mock
+  private TransactionContext context;
+
+  @Test
+  void invokeValidServiceTransaction() throws Exception {
+    ValidService service = spy(new ValidService());
+    TransactionInvoker invoker = new TransactionInvoker(service);
+    invoker.invokeTransaction(ValidService.TRANSACTION_ID, ARGUMENTS, context);
+    invoker.invokeTransaction(ValidService.TRANSACTION_ID_2, ARGUMENTS, context);
+
+    verify(service).transactionMethod(ARGUMENTS, context);
+    verify(service).transactionMethod2(ARGUMENTS, context);
+  }
+
+  @Test
+  void invokeInvalidTransactionId() {
+    TransactionInvoker invoker = new TransactionInvoker(new ValidService());
+    int invalidTransactionId = Integer.MAX_VALUE;
+    IllegalArgumentException e = assertThrows(IllegalArgumentException.class,
+        () -> invoker.invokeTransaction(invalidTransactionId, ARGUMENTS, context));
+    assertThat(e.getMessage())
+        .contains(String.format("No method with transaction id (%s)", invalidTransactionId));
+  }
+
+  @Test
+  void invokeThrowingTransaction() {
+    TransactionInvoker invoker = new TransactionInvoker(new ThrowingService());
+    TransactionExecutionException e = assertThrows(TransactionExecutionException.class,
+        () -> invoker.invokeTransaction(ThrowingService.TRANSACTION_ID, ARGUMENTS, context));
+    assertThat(e.getErrorCode()).isEqualTo(ThrowingService.ERROR_CODE);
+  }
+
```

Comment:
Please test handling of 'other' _service originated_ exceptions, since it is important to handle them properly. 

May add a second method to 'ThrowingService'.